src/generic/scrlwing.cpp

```diff
@@ -561,6 +561,12 @@ void wxScrollHelperBase::HandleOnScroll(wxScrollWinEvent& event)
     {
         m_targetWindow->ScrollWindow(dx, dy, GetScrollRect());
     }
+#ifdef __WXUNIVERSAL__
+    if (m_win != m_targetWindow)
+    {
+        m_win->Refresh(true, GetScrollRect());
```

Comment:
Could this be done from `wxWindow::SetScrollPos()` instead? It looks like the real problem is that this function doesn't update the window, unlike in the native ports.

Reply:
Maybe, I just don't understand how to do it or how native port works. I mean on level wxScrollHelperBase there is m_win and there is m_targetWindow, but on level wxWindow how they related?

Reply:
`m_win` is the window with the scrollbars, while `m_targetWindow` is the window actually being scrolled. Usually they're one and the same, but `wxGrid` is an example of when it isn't, because we want to scroll only the grid data and not the entire window when the scrollbars are used.

So `SetScrollPos()` must update the scrollbar itself, because it's always part of `m_win` -- but apparently doesn't do it in wxUniv.

Reply:
Yes, but scrollbar itself updated correctly, only grid labels re-drawing incorrectly. wxGrid labels wasn't updated correctly because there isn't Update calls for m_win (wxGrid). And it's important because grid labels it is separate widget that [scroll](https://github.com/wxWidgets/wxWidgets/blob/master/src/generic/grid.cpp#L2594) together with grid.
So instead of m_win->Refresh inside wxScrollHelperBase::HandleOnScroll we can just add calling Update in the end of wxGrid::ScrollWindow for repainting area that already [refreshed](https://github.com/wxWidgets/wxWidgets/blob/master/src/univ/winuniv.cpp#L1084) inside ScrollWindow. Does it make sense?

Reply:
The [screenshot](https://i.imgur.com/MAH4TrF.jpg) from #19357 showed corrupted scrollbar, so this is why I thought that the problem was with updating it...

Anyhow, scrolling `m_targetWindow` should _not_ require updating the window doing the scrolling. I.e. imagine that you don't use `wxScrolledWindow` at all and just call `wxWindow::ScrollWindow()` on some window in your code -- surely this should just work, without any extra `Refresh()` calls. If it doesn't work in wxUniv, it would be better to fix this, as it can/will affect other code, not just this case.

Reply:
Sorry if I didn't write it clear in first message, #19357 is indeed corrupted scrollbar but it's just regression. If revert commit https://github.com/wxWidgets/wxWidgets/commit/3b3169fa156215251164f408d9e6797d4c350725 (what i did in this PR in winuniv.cpp file) scrollbar works again for all simple widget. But problem with grid labels returns so changes in this PR from scrlwing.cpp directed to fix it.